src/App.jsx

```diff
@@ -14,45 +18,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const goodsWithProperty = goodsFromServer.map((good, i) => ({
+  id: i + 1,
+  name: good,
+  length: good.length,
+}));
+
+const SORT_BY_NAME = 'name';
+const SORT_BY_LENGTH = 'length';
+
+function prepeareGoods(goods, sortName, reverse) {
+  let prepearedGoods = [...goods];
+
+  if (sortName) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortName) {
+        case SORT_BY_NAME:
+          return good1[SORT_BY_NAME].localeCompare(good2[SORT_BY_NAME]);
+
+        case SORT_BY_LENGTH:
+          return good1[SORT_BY_LENGTH] - good2[SORT_BY_LENGTH];
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    prepearedGoods = prepearedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortedGoodsList = prepeareGoods(
+    goodsWithProperty,
+    sortField,
+    isReversed,
+  );
+
+  const ResetButton = () => (
+    <button
+      type="button"
+      className="button is-danger is-light"
+      onClick={() => {
```

Comment:
```suggestion
      onClick={reset}
```
Consider moving this logic to a separate function from jsx